nilmtk/api.py

```diff
@@ -241,60 +242,108 @@ def train_jointly(self,clf,d):
+	                test.set_window(start=d[dataset]['buildings'][building]['start_time'],end=d[dataset]['buildings'][building]['end_time'])
+	                test_mains=next(test.buildings[building].elec.mains().load(physical_quantity='power', ac_type=self.power['mains'], sample_period=self.sample_period))
+	                appliance_readings=[]
+
+	                for appliance in self.appliances:
+	                    test_df=next((test.buildings[building].elec[appliance].load(physical_quantity='power', ac_type=self.power['appliance'], sample_period=self.sample_period)))
+	                    appliance_readings.append(test_df)
+
+	                
+	                if self.DROP_ALL_NANS:
+	                    test_mains, appliance_readings = self.dropna(test_mains, appliance_readings)
+
+
+	                if self.artificial_aggregate:
+	                    print ("Creating an Artificial Aggregate")
+	                    test_mains = pd.DataFrame(np.zeros(appliance_readings[0].shape),index = appliance_readings[0].index,columns=appliance_readings[0].columns)
+	                    for app_reading in appliance_readings:
+	                        test_mains+=app_reading
+
+
+	                self.test_mains = [test_mains]
+	                for i, appliance_name in enumerate(self.appliances):
+	                    self.test_submeters.append((appliance_name,[appliance_readings[i]]))
+	                
+	                self.storing_key = str(dataset) + "_" + str(building) 
+	                self.call_predict(self.classifiers)
+
+    # def sitetest_jointly(self,d):
+
+
+    #     # store the test_main readings for all buildings
+    #     for dataset in d:
+    #         print("Loading data for ",dataset, " dataset")
+    #         test=DataSet(d[dataset]['path'])
+    #         for building in d[dataset]['buildings']:
+    #             test.set_window(start=d[dataset]['buildings'][building]['start_time'],end=d[dataset]['buildings'][building]['end_time'])
+    #             test_mains=next(test.buildings[building].elec.mains().load(physical_quantity='power', ac_type='apparent', sample_period=self.sample_period))
+    #             # print(f'test_mains: {test_mains}')
 
+                
+    #             if self.DROP_ALL_NANS:
+    #                 test_mains= self.dropna(test_mains)
 
-                self.test_mains = [test_mains]
-                for i, appliance_name in enumerate(self.appliances):
-                    self.test_submeters.append((appliance_name,[appliance_readings[i]]))
                 
-                self.storing_key = str(dataset) + "_" + str(building) 
-                self.call_predict(self.classifiers)
+
+    #             self.test_mains = [test_mains]
+    #             self.storing_key = str(dataset) + "_" + str(building) 
+    #             df = self.call_predict(self.classifiers)
+    #     return df
             
 
-    def dropna(self,mains_df, appliance_dfs):
+    def dropna(self,mains_df, appliance_dfs=None):
```

Comment:
Why not `appliance_dfs=[]` instead? Using an empty list, you don't have to change the function because all the `for` loops will exit without doing anything ;).
Then, you can just call the method like:
```python
test_mains, _ = self.dropna(mains_df, [])
```